Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -162,7 +162,17 @@ template <class Type> class TableColumn : public API::Column {
         typename boost::mpl::if_c<boost::is_convertible<double, Type>::value,
                                   Type, InconvertibleToDoubleType>::type
             DoubleType;
-    return boost::numeric_cast<double, DoubleType>(m_data[i]);
+    if (typeid(m_data[i]) == typeid(std::string)) {
+      std::string trimmedString = boost::lexical_cast<std::string>(
+          m_data[i]); // so not to alter the original data
+
+      if (trimmedString.back() == ':') {
```

Comment:
Hmm. It seems too low-level for that type of fix here (unless `TableColumn` is the one responsible for putting it there in the first place). It feels wrong that `TableColumn` has to understand the the string might contain any random characters.